raiden/api/python.py

```diff
@@ -759,12 +759,12 @@ def transfer_async(
                 raise InvalidSecretOrSecretHash('secret_hash is not an hexadecimal string.')
             secret_hash = to_bytes(hexstr=secret_hash)
 
-        if secret is None and secret_hash is not None:
-            raise InvalidSecretOrSecretHash('secret_hash without a secret is not supported yet.')
-
         if secret is not None and secret_hash is not None and secret_hash != sha3(secret):
             raise InvalidSecretOrSecretHash('provided secret and secret_hash do not match.')
 
+        if secret is None and secret_hash is not None:
+            pass
```

Comment:
Yes, This can be removed. However, I placed it in to make sure that any reader will understand that this is a valid option. Let me know if you like me to remove it.